src/ops.rs

```diff
@@ -115,7 +98,7 @@ impl IntOp {
     }
   }
 
-  fn op<T: Num>(self, a: T, b: T) -> T {
+  fn op<T: Numeric + PartialEq + PartialOrd>(self, a: T, b: T) -> T {
```

Comment:
minor, but any reason these aren't supertraits of `Numeric`? would make this line and the next a bit simpler